python/usb_console/radio_session.py

```diff
@@ -45,169 +51,53 @@ def __init__(self, device_name, datastore, logger, radio_keys_config):
-
-                            if part.get('Content-Disposition') is None:
-                                continue
-                            
-                            #get the body text of the email and look for the MOMSN/MTMSN number
-                            if part.get_content_type() == "text/plain":
-                                email_body = part.get_payload(decode=True).decode('utf8')
-                                for line in email_body.splitlines():
-                                    if line.find("MOMSN")!=-1:
-                                        self.momsn=int(line[7:])
-                                    if line.find("MTMSN")!=-1:
-                                        self.confirmation_mtmsn=int(line[7:])
-                                        if self.confirmation_mtmsn != self.mtmsn:
-                                            #stop radio session from sending any more uplinks
-                                            self.send_uplinks=False
-                                        else:
-                                            #allow radio session to send more uplinks
-                                            self.stop_uplinks=True
-                                            
-
-                            #check if there is an email attachment
-                            if part.get_filename() is not None:
-                                #get data from email attachment
-                                attachmentContents=part.get_payload(decode=True).decode('utf8')
-                                data=self.process_downlink_packet(attachmentContents)
-                                
-                                #iterate through each statefield key of the dictionary
-                                entry={}
-                                for key in self.statefields:
-                                    #if the value for a statefield exists in the json object recieved, update the dictionary and add entry to datastore
-                                    if data.get(key)is not None:
-                                        self.statefields[key]=data[key]
-                                        entry['field'] = key
-                                        entry['val'] = data[key]
-                                        entry['time'] = datetime.datetime.now()
-                                        self.datastore.consume_queue_item(entry)
-
     def read_state(self, field, timeout=None):
         '''
-        Read state.
-        
-        Read the latest value of the state field received via downlink with the name "field".
+        Read state by posting a request for data to the Flask server
         '''
-        return self.statefields[field]
+        headers = {
+            'Accept': 'text/html',
+        }
+
+        response = requests.get('http://'+self.flask_server+':'+self.flask_port+'/search-statefields?imei='+str(self.imei)+'&field='+str(field), headers=headers)
```

Comment:
This kind of string building will fail if the inputs are not correctly URL-encoded. For example, if `field` contains a period, the request URL will be malformed.

There is a cleaner and safer way to do this within `requests`. See "Making a POST request" on https://www.geeksforgeeks.org/get-post-requests-using-python/